include/ringmesh/mesh/cartesian_grid.h

```diff
@@ -63,50 +71,99 @@ namespace RINGMesh
             ReferenceFrame< DIMENSION > vec_cartesian_axis )
             : nb_cells_in_each_direction_(
                   std::move( nb_cells_in_each_direction ) ),
-              cartesian_frame_( std::move( vec_cartesian_axis ) ),
               nb_total_cells_( 1 ),
-              inverse_cartesian_frame_( ReferenceFrameManipulator::
+              cartesian_frame_( std::move( vec_cartesian_axis ) ),
+              inverse_cartesian_frame_( ReferenceFrameManipulator< DIMENSION >::
                       reference_frame_from_global_to_local( cartesian_frame_ ) )
         {
+            if( !ReferenceFrameManipulator< DIMENSION >::frame_is_orthogonal(
+                    cartesian_frame_ ) )
+            {
+                throw RINGMeshException( "RINGMesh Test",
+                    "Warning : the frame of the Cartesian Grid "
+                    "you're trying to create is not orthogonal " );
+            }
             for( auto i : range( DIMENSION ) )
             {
+                if( nb_cells_in_each_direction[i] < 1 )
+                {
+                    throw RINGMeshException( "RINGMesh Test",
+                        "Warning : You're trying to create a Cartesian Grid "
+                        "with no cell in direction ",
+                        i, ", and Cartesian Grid must have at least one cell "
+                           "in each direction" );
+                }
                 nb_total_cells_ *= nb_cells_in_each_direction[i];
             }
+            attributes_manager_.resize( nb_total_cells_ );
         }
 
-        static MeshType type_name_static()
-        {
-            return "CartesianGrid";
-        }
         void save_mesh( const std::string& filename ) const
         {
             // TODO
         }
-        void print_mesh_bounded_attributes() const
+
+        //        void print_mesh_bounded_attributes( const std::string&
+        //        output_location ) const
+        //        {
+        //        	print_bounded_attributes( attributes_manager_,
+        //        output_location );
+        //        }
+
+        void resize( ivecn< DIMENSION >& new_size,
```

Comment:
Is it really a resize or a complete new grid? When could you need to resize this kind of grid?

Reply:
It is a resize, the attributes already set there should stay the same.
I have no idea yet what it could be useful for, but it may make possible the merging of several Cartesian Grids (after a parallel computation on each of them) while not having to copy some of the information. This is just a supposition though, and that's why I give out a warning if the user wants to call this method, so he is aware of it.

Reply:
Actually, after talking with a potential user of the cartesian grids, it appears this method would be needed a lot for his applications.